app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_create :increment_user_posts_count
+
+  def increment_user_posts_count
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] Nice work using the rails callback method. However, you will need the `after_save` method instead of the `after_create` method here. It is also advisable to make the `increment_user_posts_count` method private. when using the rails active callbacks `after_save` method to call it. :+1:

Reply:
done